### PR TITLE
Require Kokkos 4.1

### PR DESCRIPTION
**Pull request by aprokop**

### CODE REVIEW

File: docker/Dockerfile.hipcc [hunk 1]
```diff
@@ -89,7 +89,7 @@ RUN SCRATCH_DIR=/scratch && mkdir -p ${SCRATCH_DIR} && cd ${SCRATCH_DIR} && \
     rm -rf ${SCRATCH_DIR}
 
 # Install Kokkos
-ARG KOKKOS_VERSION=4.0.00
```
Comment (masterleinad) [inline docker/Dockerfile.hipcc:92]: Why not bump `4.0.00` to `4.1.00` and `4.1.00` to `4.2.00`?

Comment (aprokop) [reply docker/Dockerfile.hipcc:92]: I thought we would leap frog the versions. Change 4.0 to 4.2 now, change 4.1 to 4.3 next time. This way we have a mix of older builds and newer builds.

File: src/ArborX_LinearBVH.hpp [hunk 2]
```diff
@@ -32,6 +31,7 @@
 #include <ArborX_TraversalPolicy.hpp>
 
 #include <Kokkos_Core.hpp>
+#include <Kokkos_Profiling_ScopedRegion.hpp>
```
Comment (dalg24) [inline src/ArborX_LinearBVH.hpp:34]: I just realized it is not provided by `<Kokkos_Core.hpp>` :(

File: src/details/ArborX_MinimumSpanningTree.hpp [hunk 3]
```diff
@@ -13,7 +13,8 @@
 #define ARBORX_MINIMUM_SPANNING_TREE_HPP
 
 #include <ArborX_AccessTraits.hpp>
-#include <ArborX_DetailsKokkosExtScopedProfileRegion.hpp>
+#include <ArborX_DetailsKokkosExtArithmeticTraits.hpp>
+#include <ArborX_DetailsKokkosExtMinMaxOperations.hpp>
```
Comment (dalg24) [inline src/details/ArborX_MinimumSpanningTree.hpp:17]: What is the deal with these?  Oversight?

File: src/details/ArborX_MinimumSpanningTree.hpp [hunk 4]
```diff
@@ -22,6 +23,9 @@
 #include <ArborX_DetailsWeightedEdge.hpp>
 #include <ArborX_LinearBVH.hpp>
 
+#include <Kokkos_Core.hpp>
```
Comment (dalg24) [inline src/details/ArborX_MinimumSpanningTree.hpp:26]: OK

File: src/details/ArborX_MinimumSpanningTree.hpp (hunk 3 above)
Comment (aprokop) [reply src/details/ArborX_MinimumSpanningTree.hpp:17]: Yes, probably happened during a rebase.

File: docker/README.md [hunk 5]
```diff
@@ -3,7 +3,7 @@ in the environment file.  Its value will be prepended along with the service
 name to the container on start up.
 
 
-You may use multiple compose files to customize your container.  For instance, you
+You may use multiple compose files to customize your container. For instance, you
```
Comment (dalg24) [inline docker/README.md:6]: Why did you change?  This is likely me using two spaces between sentences.
In any case, not really related to this PR.  I am not requesting changes but if there is any further push I would like you to undo.

**Review by dalg24 — APPROVED** [review]

File: docker/README.md (hunk 5 above)
Comment (aprokop) [reply docker/README.md:6]: I changed it in [a9dba00](https://github.com/arborx/ArborX/commit/a9dba003b35ec6c8ba2ca155ac81a45dd085d791), before your review, when I needed to force push to fix the incorrect Jenkins links, and was looking for a minor change.
